com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/QuickOutline.java

```diff
@@ -0,0 +1,344 @@
+            }
+        });
+        return treeViewer.getControl();
+    }
+
+    @Override
+    public void setInput(Object input) {
+        treeViewer.setInput(input);
+    }
+
+    @Override
+    public boolean hasContents() {
+        if (treeViewer == null || treeViewer.getInput() == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void setInformation(String information) {
+    }
+
+    @Override
+    public void setSizeConstraints(int maxWidth, int maxHeight) {
+        getShell().setSize(maxWidth, maxHeight);
+    }
+
+    @Override
+    public Point computeSizeHint() {
+        return getShell().getSize();
+    }
+
+    @Override
+    public void setVisible(boolean visible) {
+        if (visible) {
+            open();
+        } else {
+            saveDialogBounds(getShell());
+            getShell().setVisible(false);
+        }
+    }
+
+    @Override
+    public void setSize(int width, int height) {
+        getShell().setSize(width, height);
+    }
+
+    @Override
+    public void setLocation(Point location) {
+        getShell().setLocation(location);
+    }
+
+    @Override
+    public void dispose() {
+        close();
+    }
+
+    @Override
+    public void addDisposeListener(DisposeListener listener) {
+        getShell().addDisposeListener(listener);
+    }
+
+    @Override
+    public void removeDisposeListener(DisposeListener listener) {
+        getShell().removeDisposeListener(listener);
+    }
+
+    @Override
+    public void setForegroundColor(Color foreground) {
+        applyForegroundColor(foreground, getContents());
+    }
+
+    @Override
+    public void setBackgroundColor(Color background) {
+        applyBackgroundColor(background, getContents());
+    }
+
+    @Override
+    public boolean isFocusControl() {
+        return treeViewer.getTree().isFocusControl() || filterText.isFocusControl();
+    }
+
+    @Override
+    public void setFocus() {
+        getShell().forceFocus();
+        if (filterText != null) {
+            filterText.setFocus();
+        }
+    }
+
+    @Override
+    public void addFocusListener(FocusListener listener) {
+        getShell().addFocusListener(listener);
+    }
+
+    @Override
+    public void removeFocusListener(FocusListener listener) {
+        getShell().removeFocusListener(listener);
+    }
+
+    private boolean matchesFilter(Object element) {
+        if (treeViewer == null) {
+            return true;
+        }
+
+        if (element instanceof OutlineElement) {
+            String matchName = ((OutlineElement) element).getText();
+            String text = filterText.getText();
+
+            if (Strings.emptyToNull(text) == null) {
+                return true;
+            }
+
+            if (matchName != null) {
+                return matchName.contains(text);
+            }
+        }
+
+        return false;
+    }
+
+    public class NamePatternFilter extends ViewerFilter {
```

Comment:
I wonder why a combination of `PatternFilter` + `FilteredTree` (instead of `Tree`) was not appropriate here. I am sure you had reasons to choose a `ViewerFilter`, just wondering why